flink-libraries/flink-table/src/main/scala/org/apache/flink/table/sinks/TableSink.scala

```diff
@@ -45,6 +45,10 @@ trait TableSink[T] {
   /** Returns the types of the table fields. */
   def getFieldTypes: Array[TypeInformation[_]]
 
+  /** Describes the table sink. */
+  def explainSink(): String =
```

Comment:
This breaks compatibility with TableSinks implemented in Java. I'm fine with this. What do you think @fhueske?

Reply:
As far as I can see, this method is only called from within the interface implementations and not from external code. I'd rather not add this method unless it is required.